go/border/rpkt/route.go

```diff
@@ -148,36 +150,94 @@ func (rp *RtrPkt) forwardFromExternal() (HookResult, *common.Error) {
 	}
 	// If this is a cross-over Hop Field, increment the path.
 	if rp.hopF.Xover {
-		if err := rp.IncPath(); err != nil {
-			return HookError, err
-		}
-		// FIXME(kormat): this might need to change when multiple interfaces
-		// per router are supported.
-		if err := rp.validatePath(DirLocal); err != nil {
+		if err := rp.xoverFromExternal(); err != nil {
 			return HookError, err
 		}
 	}
 	// Destination is in a remote ISD-AS, so forward to egress router.
 	// FIXME(kormat): this will need to change when multiple interfaces per
 	// router are supported.
-	nextIF, err := rp.IFNext()
-	if err != nil {
-		return HookError, err
-	}
-	if err := rp.validateLocalIF(*nextIF); err != nil {
-		return HookError, err
-	}
-	nextBR := conf.C.TopoMeta.IFMap[int(*nextIF)]
+	nextBR := conf.C.TopoMeta.IFMap[int(*rp.ifNext)]
 	dst := &net.UDPAddr{IP: nextBR.BasicElem.Addr.IP, Port: nextBR.BasicElem.Port}
 	rp.Egress = append(rp.Egress, EgressPair{callbacks.locOutFs[intf.LocAddrIdx], dst})
 	return HookContinue, nil
 }
 
+// xoverFromExternal handles XOVER hop fields at the ingress router, including
+// a lot of sanity/security checking.
+func (rp *RtrPkt) xoverFromExternal() *common.Error {
+	infoF := rp.infoF
+	origIFCurr := *rp.ifCurr
+	origIFNext := *rp.ifNext
+	var segChgd bool
+	var err *common.Error
+	if segChgd, err = rp.IncPath(); err != nil {
```

Comment:
I'm not sure about the semantics of Go. Will segChgd, err be set for only inside the "if" code block, or also for the code below? (It's important, since we check if segChgd is set, further below)